### PR TITLE
Simplify ansible tasks to deploy / undeploy target cluster

### PR DESCRIPTION
**Pull request by maelk**
This commit removes the kustomizations used to split the monolithic
deployment and instead uses separate deployments for each item

### CODE REVIEW

File: vm-setup/roles/v1aX_integration_test/tasks/provision_controlplane.yml [hunk 1]
```diff
@@ -1,41 +1,7 @@
...
-# Provision cluster & master node
-  - name: Provision Ubuntu based controlplane node
-    block:
-      - name: Generate template for v1alpha3
-        include: generate_templates.yml
+  - name: Create control plane
+    k8s:
+      state: present
+      src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_controlplane_{{ IMAGE_OS|lower }}.yaml"
+      namespace: "{{ NAMESPACE }}"
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/provision_controlplane.yml:7]: Remove the white space please.

File: vm-setup/roles/v1aX_integration_test/tasks/provision_cluster.yml [hunk 2]
```diff
@@ -1,30 +1,7 @@
...
-      src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_cluster_{{ CLUSTER_NAME }}_{{ IMAGE_OS|lower }}.yaml"
-      dest: "/tmp/cluster.yaml"
-
   - name: Provision cluster
     k8s:
       state: present
-      src: "/tmp/cluster.yaml"
+      src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_cluster_{{ IMAGE_OS|lower }}.yaml"
       namespace: "{{ NAMESPACE }}"
     register: provision_cluster
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/provision_cluster.yml:7]: Could you please remove the register if it is not being used?

File: vm-setup/roles/v1aX_integration_test/tasks/deprovision_controlplane.yml [hunk 3]
```diff
@@ -1,33 +1,7 @@
...
-      - name: Deprovision CentOS master node.
-        k8s:
-          state: absent
-          src: "/tmp/controlplane_centos.yaml"
-          namespace: "{{ NAMESPACE }}"
-        ignore_errors: yes
-        register: deprovision_centos_master_node
-    when: (IMAGE_OS == "Centos") or
-          (IMAGE_OS == "")
+  - name: Deprovision Ubuntu master node.
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/deprovision_controlplane.yml:2]: The name of the task is misleading as the same task used for both Ubuntu & Centos.

File: vm-setup/roles/v1aX_integration_test/tasks/deprovision_worker.yml [hunk 4]
```diff
@@ -1,33 +1,7 @@
...
-      - name: Deprovision CentOS worker node.
-        k8s:
-          state: absent
-          src: "/tmp/workers_centos.yaml"
-          namespace: "{{ NAMESPACE }}"
-        ignore_errors: yes
-        register: deprovision_centos_worker_node
-    when: (IMAGE_OS == "Centos") or
-          (IMAGE_OS == "")
+  - name: Deprovision Ubuntu worker node.
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/deprovision_worker.yml:2]: Same here with naming.

File: vm-setup/roles/v1aX_integration_test/tasks/provision_worker.yml [hunk 5]
```diff
@@ -1,41 +1,6 @@
...
-
-      - name: Create CentOS worker node
-        k8s:
-          state: present
-          src: "/tmp/workers_centos.yaml"
-          namespace: "{{ NAMESPACE }}"
-        register: centos_worker_node
-    when: (IMAGE_OS == "Centos") or
-          (IMAGE_OS == "")
+  - name: Create Ubuntu worker node
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/provision_worker.yml:2]: Same here with naming.

**Comment by maelk** [comment]
thanks @fmuyassarov I have addressed the comment and removed all separate files.
/test-integration

File: vm-setup/roles/v1aX_integration_test/tasks/main.yml [hunk 6]
```diff
@@ -1,78 +1,89 @@
...
-  with_items:
-    - download_image
-    - provision_worker
-  when: v1aX_integration_test_action == "provision_worker"
+- name: Create worker nodes
+  k8s:
+    state: present
+    src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_workers_{{ IMAGE_OS|lower }}.yaml"
+    namespace: "{{ NAMESPACE }}"
+    when: v1aX_integration_test_action in provision_workers_actions
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/main.yml:49]: Ops, there is a white space before the condition.

Comment (maelk) [reply vm-setup/roles/v1aX_integration_test/tasks/main.yml]: typo alert! fixed, thank you

**Pull request merged**